### PR TITLE
[DiagnosticRP]add a new cli extension for spot vm placement score

### PR DESCRIPTION
**Pull request by joal42**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
Example cli: az compute diagnostic spot-placement-recommender generate -l eastus --subscription ffffffff-ffff-ffff-ffff-ffffffffffff --availability-zones true --desired-locations '["eastus", "eastus2"]' --desired-count 1 --desired-sizes '[{"sku": "Standard_D2_v2"}]'
 
aaz pr link: https://github.com/Azure/aaz/pull/436

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/compute-diagnostic-rp/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Azure CLI ComputeDiagnosticRp Extension #
+This is an extension to Azure CLI to manage ComputeDiagnosticRp resources.
+
+## How to use ##
+Please add commands usage here.
```
Comment (necusjz) [inline src/compute-diagnostic-rp/README.md:5]: please complete this part

**Comment by necusjz** [comment]
please provide the link of your pr in aaz repo

**Comment by joal42** [comment]
> please provide the link of your pr in aaz repo

https://github.com/Azure/aaz/pull/436

File: src/compute-diagnostic-rp/README.md (hunk 1 above)
Comment (joal42) [reply src/compute-diagnostic-rp/README.md]: Added.

**Review by necusjz — APPROVED** [review]

**Pull request merged**